### PR TITLE
frontend: mapping template type to fields 

### LINKED ISSUE #2110: Dashboard becomes blank after creating schedule
## Bug Report

**What version of Kubernetes are you using?**
client:v1.21.2
server:v1.21.2

**What version of Chaos Mesh are you using?**
```
Controller manager Version: version.Info{GitVersion:"master-g2bc0fa81544659", GitCommit:"2bc0fa815446599852116ac0dfb612f659376745", BuildDate:"2021-07-12T09:58:17Z", GoVersion:"go1.15.11", Compiler:"gc", Platform:"linux/amd64"}
```

**What did you do?**
After creating a schedule, the whole page will be a blank, and go back also doesn't work. I must re-enter the address to see the dashboard. And I still can not get details after clicking the schedule I created.

**What did you expect to see?**
See the schedule page.

### PR DESCRIPTION
**Pull request by STRRL**
STRRL <str_ruiling@outlook.com>

### What problem does this PR solve?

Issue Number: close #2110

Problem Summary:

- frontend uses the bad field name when fetching spec from EmbedChaos

### What is changed and how it works?

What's Changed:

- use code generation to generate the mapping 

### Related changes

* PR to update `chaos-mesh/website`/`chaos-mesh/website-zh`:
* Need to update Chaos Dashboard component, related issue:
* Need to cheery-pick to the release branch

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Breaking backward compatibility

### Release note 

```release-note
Please add a release note.
If you don't think this PR needs a release note then fill it with None.
```

### CODE REVIEW

File: ui/src/components/ObjectConfiguration/index.tsx [hunk 1]
```diff
@@ -31,7 +31,9 @@ const ObjectConfiguration: React.FC<ObjectConfigurationProps> = ({
 
   const spec = inNode ? config : config.kube_object.spec
   const experiment =
-    inSchedule || inNode ? spec[toCamelCase(inSchedule ? spec.type : (config as any).templateType)] : spec
+    inSchedule || inNode
+      ? spec[mappingTemplateType2FieldName(inSchedule ? spec.type : (config as any).templateType)]
```
Comment (g1eny0ung) [inline ui/src/components/ObjectConfiguration/index.tsx:35]: Should be `mappingTemplateType2FieldName.get`.

Comment (g1eny0ung) [reply ui/src/components/ObjectConfiguration/index.tsx:35]: I also think should give it a meaningful name, like `kindToTemplateType`?

Comment (STRRL) [reply ui/src/components/ObjectConfiguration/index.tsx:35]: > Should be `mappingTemplateType2FieldName.get`.

`mappingTemplateType2FieldName` is the name of function, it call `mapping.get()` inside.

Comment (STRRL) [reply ui/src/components/ObjectConfiguration/index.tsx:35]: > I also think should give it a meaningful name, like `kindToTemplateType`?

How about `TemplateTypeToFieldName`?

Comment (g1eny0ung) [reply ui/src/components/ObjectConfiguration/index.tsx:35]: Oh, got it. I miss the function definition. `TemplateTypeToFieldName` is OK.

Comment (STRRL) [reply ui/src/components/ObjectConfiguration/index.tsx]: Oh @g1eny0ung  which should I use, `TemplateTypeToFieldName` or `templateTypeToFieldName`?

Comment (g1eny0ung) [reply ui/src/components/ObjectConfiguration/index.tsx]: The latter. `camelCase` is the customary usage of JS.

Comment (STRRL) [reply ui/src/components/ObjectConfiguration/index.tsx]: updated PTAL @g1eny0ung

**Review by g1eny0ung — APPROVED** [review]
LGTM

**Review by cwen0 — APPROVED** [review]
LGTM

**Pull request merged**